compiler/luci/env/src/UserSettings.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/UserSettings.h"
+
+namespace luci
+{
+
+class UserSettingsImpl : public UserSettings
+{
+public:
+  void set(const Key &key, bool value) override;
+  bool get(const Key &key) const override;
+
+private:
+  bool _MuteWarnings{false};
+  bool _DisableValidation{false};
+};
+
+void UserSettingsImpl::set(const Key &key, bool value)
+{
+  switch (key)
+  {
+    case Key::MuteWarnings:
+      _MuteWarnings = value;
+      break;
+    case Key::DisableValidation:
+      _DisableValidation = value;
+      break;
+    default:
+      break;
```

Comment:
(Optional) How about throwing `runtime_error` if invalid key is given?

Reply:
seems reasonable.